test/integration/controller_instance_test.go

```diff
@@ -0,0 +1,1475 @@
+				setup: func(ct *controllerTest) {
+					ct.osbClient.DeprovisionReaction = &fakeosb.DeprovisionReaction{
+						Response: tc.osbResponse,
+					}
+				},
+			}
+			ct.run(tc.testFunction)
+		})
+	}
+}
+
+func TestPollServiceInstanceLastOperation(t *testing.T) {
+	cases := []struct {
+		name                         string
+		setupFunc                    func(t *controllerTest)
+		skipVerifyingInstanceSuccess bool
+		skipInstanceDelete           bool
+		verifyCondition              v1beta1.ServiceInstanceCondition
+		preDeleteBroker              func(t *controllerTest)
+		preCreateInstance            func(t *controllerTest)
+		postCreateInstance           func(t *controllerTest)
+	}{
+		{
+			name: "provisioning with last operation response state in progress",
+			setupFunc: func(ct *controllerTest) {
+				ct.osbClient.ProvisionReaction = &fakeosb.ProvisionReaction{
+					Response: &osb.ProvisionResponse{
+						Async: true,
+					},
+				}
+				ct.osbClient.PollLastOperationReaction = &fakeosb.PollLastOperationReaction{
+					Response: &osb.LastOperationResponse{
+						State:       osb.StateInProgress,
+						Description: strPtr("testDescription"),
+					},
+				}
+			},
+			skipVerifyingInstanceSuccess: true,
+			skipInstanceDelete:           true,
+			verifyCondition: v1beta1.ServiceInstanceCondition{
+				Type:   v1beta1.ServiceInstanceConditionReady,
+				Status: v1beta1.ConditionFalse,
+				Reason: "Provisioning",
+			},
+		},
+		{
+			name: "provisioning with last operation response state succeeded",
+			setupFunc: func(ct *controllerTest) {
+				ct.osbClient.ProvisionReaction = &fakeosb.ProvisionReaction{
+					Response: &osb.ProvisionResponse{
+						Async: true,
+					},
+				}
+				ct.osbClient.PollLastOperationReaction = &fakeosb.PollLastOperationReaction{
+					Response: &osb.LastOperationResponse{
+						State:       osb.StateSucceeded,
+						Description: strPtr("testDescription"),
+					},
+				}
+			},
+			verifyCondition: v1beta1.ServiceInstanceCondition{
+				Type:   v1beta1.ServiceInstanceConditionReady,
+				Status: v1beta1.ConditionTrue,
+				Reason: "ProvisionedSuccessfully",
+			},
+		},
+		{
+			name: "provisioning with last operation response state failed",
+			setupFunc: func(ct *controllerTest) {
+				ct.osbClient.ProvisionReaction = &fakeosb.ProvisionReaction{
+					Response: &osb.ProvisionResponse{
+						Async: true,
+					},
+				}
+				ct.osbClient.PollLastOperationReaction = &fakeosb.PollLastOperationReaction{
+					Response: &osb.LastOperationResponse{
+						State:       osb.StateFailed,
+						Description: strPtr("testDescription"),
+					},
+				}
+			},
+			skipVerifyingInstanceSuccess: true,
+			verifyCondition: v1beta1.ServiceInstanceCondition{
+				Type:   v1beta1.ServiceInstanceConditionReady,
+				Status: v1beta1.ConditionFalse,
+				Reason: "ProvisionCallFailed",
+			},
+		},
+		{
+			name: "last operation response successful with originating identity",
```

Comment:
I don't see much value in this test case. The last operation poll behavior is not dependent in any way on originating identity. I recommend we remove the test case.

Reply:
I was trying to replicate the unit test called TestPollInstanceUsingOriginatingIdentity. As I said above, perhaps this entire test needs deleting.